main/client/src/mill/main/client/MillClientMain.java

```diff
@@ -209,9 +211,10 @@ public static int run(String lockBase,
 
         while (ioSocket == null && System.currentTimeMillis() - retryStart < 5000) {
             try {
+                String socketBaseName = "mill." + md5hex(new File(lockBase).getCanonicalPath());
                 ioSocket = Util.isWindows?
-                        new Win32NamedPipeSocket(Util.WIN32_PIPE_PREFIX + "mill." + new File(lockBase).getName())
-                        : new UnixDomainSocket(lockBase + "/io");
+                        new Win32NamedPipeSocket(Util.WIN32_PIPE_PREFIX + socketBaseName)
+                        : new UnixDomainSocket(socketBaseName + "/io");
```

Comment:
I think on non-Windows systems we still need to provide an proper path to `UnixDomainSocket`, were all parent directories exists. After your change, this is no longer the case.

Reply:
Yep, figured after tests failed.. 😄  
Should be fixed now.